### DISCUSSION TITLE
Limitations with Python GraphQL Client

### DISCUSSION THREAD

**Discussion by tbonnerctps** [requarks/wiki#5999] [Help / Questions] [answered]
I've attempted to create a new page and create a new user using the Python GraphQL client and ran into the following error. Do any clients allow creating new pages and/or creating new users with version 2.5.292? If not, is this going to be supported in version 3? What is the most fully-functional client available right now / what client is recommended for use? Any guidance is greatly appreciated.

Create new page response:
{'errors': [{'message': 'Forbidden', 'locations': [{'line': 3, 'column': 5}], 'path': ['pages', 'create'], 'extensions': {'code': 'INTERNAL_SERVER_ERROR', 'exception': {'stacktrace': ['Error: Forbidden', '    at field.resolve (/wiki/server/graph/directives/auth.js:47:17)', '    at field.resolve (/wiki/node_modules/apollo-server-core/dist/utils/schemaInstrumentation.js:52:26)', '    at resolveFieldValueOrError (/wiki/node_modules/graphql/execution/execute.js:502:18)', '    at resolveField (/wiki/node_modules/graphql/execution/execute.js:460:16)', '    at executeFields (/wiki/node_modules/graphql/execution/execute.js:297:18)', '    at collectAndExecuteSubfields (/wiki/node_modules/graphql/execution/execute.js:748:10)', '    at completeObjectValue (/wiki/node_modules/graphql/execution/execute.js:738:10)', '    at completeValue (/wiki/node_modules/graphql/execution/execute.js:626:12)', '    at /wiki/node_modules/graphql/execution/execute.js:527:16', '    at runMicrotasks (<anonymous>)', '    at processTicksAndRejections (node:internal/process/task_queues:96:5)']}}}], 'data': {'pages': {'create': None}}}

Create new user response:
{'errors': [{'message': 'Forbidden', 'locations': [{'line': 4, 'column': 5}], 'path': ['users', 'create'], 'extensions': {'code': 'INTERNAL_SERVER_ERROR', 'exception': {'stacktrace': ['Error: Forbidden', '    at field.resolve (/wiki/server/graph/directives/auth.js:47:17)', '    at field.resolve (/wiki/node_modules/apollo-server-core/dist/utils/schemaInstrumentation.js:52:26)', '    at resolveFieldValueOrError (/wiki/node_modules/graphql/execution/execute.js:502:18)', '    at resolveField (/wiki/node_modules/graphql/execution/execute.js:460:16)', '    at executeFields (/wiki/node_modules/graphql/execution/execute.js:297:18)', '    at collectAndExecuteSubfields (/wiki/node_modules/graphql/execution/execute.js:748:10)', '    at completeObjectValue (/wiki/node_modules/graphql/execution/execute.js:738:10)', '    at completeValue (/wiki/node_modules/graphql/execution/execute.js:626:12)', '    at /wiki/node_modules/graphql/execution/execute.js:527:16', '    at runMicrotasks (<anonymous>)', '    at processTicksAndRejections (node:internal/process/task_queues:96:5)']}}}], 'data': {'users': {'create': None}}}

**Comment by NGPixel** [comment, marked as answer]
Are you sending a valid bearer token in the `Authentication` header? The error message says "Forbidden" which means you are not.

**Reply by tbonnerctps** [reply]
Yes, I have a valid bearer token that I'm sending in the HTTP header that I generated from the API Access panel with Full Access permissions. Other operations like queries for pages work well, but a subset of mutations do not and return the Forbidden error message, even with the bearer token sent as a header.
